src/components/business/table/BusinessTable.tsx

```diff
@@ -67,6 +72,55 @@ const BusinessTable: React.FC = () => {
 
     fetchUserData();
   }, []);
+  const handleStatusChange = (newStatus: string,index :number) => {
+    setSelectedStatus(newStatus);
+    setSelectedIndex(index);
+    setDialogOpen(true);
+    
+  };
+  const handleStatusUpdate  = async () =>{
+    if (selectedIndex === -1) {
+      toast({
+        title: "Error",
+        description: "No user selected for status update.",
+        variant: "destructive",
+      });
+      return;
+    }
+  
+    const status = userData[selectedIndex].status;
```

Comment:
Unused variable

Reply:
It is being use in the catch part if there is an api error in updating then we need to retract the status to the previous status

Reply:
okay